api/lib/fiatvalues-api.js

```diff
@@ -106,19 +112,22 @@ class fiatValueAPI {
     const fiatCurrenciesUppercase = allFiatCurrencies.map((fiatCurrency) =>
       fiatCurrency.toUpperCase()
     )
-    let allFiatValues = geckoCoinIDs.reduce((allFiatValues, coinID) => {
-      const coinDenom = this.fiatValuesAPIReverseDictionary[coinID] // the actual denom of the coin
-      fiatCurrenciesUppercase.forEach((fiatCurrency) => {
-        allFiatValues = {
-          ...allFiatValues,
-          [coinDenom]: {
-            ...allFiatValues[coinDenom],
-            [fiatCurrency]: this.priceFeed[coinID][fiatCurrency.toLowerCase()]
+    let allFiatValues = {}
+    if (Object.keys(this.priceFeed).length > 0) {
```

Comment:
is this necessary? if the keys are not there `.reduce` will simply run 0 times, am I wrong?